src/Build/Construction/Solution/SolutionProjectGenerator.cs

```diff
@@ -872,10 +898,22 @@ private ProjectInstance CreateTraversalInstance(string wrapperProjectToolsVersio
 
             // Put locals second so they can override globals if they want
             traversalProject.PrependChild(importBeforeLocal);
+            traversalProject.PrependChild(directorySolutionPropsImport);
+            traversalProject.PrependChild(directorySolutionPropsPropertyGroup);
             traversalProject.PrependChild(importBefore);
             traversalProject.AppendChild(importAfter);
+            traversalProject.AppendChild(directorySolutionTargetsPropertyGroup);
+            traversalProject.AppendChild(directorySolutionTargetsImport);
             traversalProject.AppendChild(importAfterLocal);
 
+            directorySolutionTargetsPropertyGroup.AppendChild(directorySolutionTargetsFileProperty);
+            directorySolutionTargetsPropertyGroup.AppendChild(directorySolutionTargetsBasePathProperty);
+            directorySolutionTargetsPropertyGroup.AppendChild(directorySolutionTargetsPathProperty);
```

Comment:
extreme nit: personally I'd move these up to closer to where `directorySolutionTargetsPropertyGroup` is created.

Reply:
These can't be appended until their parent's have been appended